### PR TITLE
25 integrate reusable workflows

### PR DESCRIPTION
**Pull request by theissenhelen**
This PR adds reusable workflows into the CI pipeline:

- read-the-docs PR update check
- public PR label
- downstream-ci
- downstream-ci-hpc

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: ci
+
+on:
+  # Trigger the workflow on push to master or develop, except tag creation
+  push:
+    branches:
+      - 'master'
+      - 'develop'
```
Comment (JPXKQX) [inline .github/workflows/ci.yml:8]: It should be main (also in the comment above)

Comment (theissenhelen) [reply .github/workflows/ci.yml:8]: Good point.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: ci
+
+on:
+  # Trigger the workflow on push to master or develop, except tag creation
+  push:
+    branches:
+      - 'master'
```
Comment (JPXKQX) [inline .github/workflows/ci.yml:7]: There are still some "master" occurrences

File: .github/ci-config.yml
Comment (JPXKQX) [inline .github/ci-config.yml:1]: How will this file be used? I don't see any reference to it in the other files

**Review by JPXKQX — CHANGES_REQUESTED** [review]
It looks good to me. Ready to merge when there are no more "master" branches.

File: .github/ci-config.yml
Comment (theissenhelen) [reply .github/ci-config.yml:1]: This is part of the downstream-ci pipeline and provides additional input to the workflow.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (theissenhelen) [reply .github/workflows/ci.yml]: Thanks, that one sneakily got in.

**Review by JPXKQX — APPROVED** [review]
Looks good! Great work

**Pull request merged**